### ISSUE TITLE
New Package (pysam, and modbampy)

### ISSUE THREAD

**Issue by rrazaghi** [pyodide/pyodide#2111]
## 🐍 Package Request 1

- Package Name and Version : pysam
- Package URL :  https://github.com/pysam-developers/pysam
- Package Dependencies that needs to be resolved first: htslib

## 🐍 Package Request 2

- Package Name and Version : modbampy
- Package URL :  https://github.com/epi2me-labs/modbam2bed  , https://pypi.org/project/modbampy/
- Package Dependencies that needs to be resolved first: htslib

Hi,
thank you for this great project. I have been trying to build these two packages however I can't make any progress at the build (python -m pyodide_build buildpkg meta.yaml)

I would really appreciate it if you could help me with this. We can start with modbampy as it is a much much smaller package than pysam. pysam is heavily used in bioinformatics.

Thank you!

**Comment by rrazaghi** [comment]
for pysam I receive the following issue when compiling htslib: 
`htslib/hfile_libcurl.c:47:10: fatal error: 'curl/curl.h' file not found
#include <curl/curl.h>
         ^~~~~~~~~~~~~
1 error generated.`

**Comment by rrazaghi** [comment]
for modbampy I have some success adding some patches but now I'm at this:
`emcc: error: Passing any of -msse, -msse2, -msse3, -mssse3, -msse4.1, -msse4.2, -mavx, -mfpu=neon flags also requires passing -msimd128!`

what's the best way to resolve this?

**Comment by hoodmane** [comment]
It's likely it would be helpful to filter out these flags in pywasmcross.
If you return `None` from this function:
https://github.com/pyodide/pyodide/blob/039618988fe9a908b08ef4f4f649775e3fdcd988/pyodide-build/pyodide_build/pywasmcross.py#L354
it will filter out the flag.

It's probably easiest if you open a PR with your partial progress.

**Comment by rrazaghi** [comment]
Thank you @hoodmane! This is all very new to me and I'd appreciate your help.

**Comment by rrazaghi** [comment]
It would be really great to figure out htslib dependencies. Almost all of the tools written for bioinformatics use htslib. This would really expand pyodide’s use case you n the field of genomics!

**Comment by rrazaghi** [comment]
Hi,

I have made some progress on pysam package (I have been able to build htslib separately) however I am stuck at this stage:

to build pysam without the libcurl dependencies we have to install this way according to documentation:

`HTSLIB_CONFIGURE_OPTIONS='--enable-plugins --disable-libcurl' python setup.py build`

however when I try to edit the meta.yaml file this seems not to work (get `fatal error: 'curl/curl.h' file not found`):

```
package:
  name: pysam
  version: 0.18.0
source:
  url: https://files.pythonhosted.org/packages/15/79/00249a37089a6aecb70a4080f8af8d88e7005e2019c71e505c3ad3a0a018/pysam-0.18.0.tar.gz
  sha256: 1d6d49a0b3c626fae410a93d4c80583a8b5ddaacc9b46a080b250dbcebd30a59
test:
  imports:
  - pysam
requirements:
  run:
    - htslib
    - samtools
    - bcftools
build:
  script: |
    export HTSLIB_CONFIGURE_OPTIONS=--disable-libcurl
about:
  home: https://github.com/pysam-developers/pysam
  PyPI: https://pypi.org/project/pysam
  summary: pysam
  license: MIT
```

am I missing something here?

**Comment by rrazaghi** [comment]
@hoodmane for modbampy I had some success but now I'm down to this error with emcc:

```npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm ERR! missing: terser@5.10.0, required by pyodide-0.19.0
[2022-01-22 13:43:39] Building package modbampy...
fatal: not a git repository: /home/roham/software/pyodide-0.19.0/packages/modbampy/build/modbampy-0.4.0/htslib/../.git/modules/htslib
fatal: not a git repository: /home/roham/software/pyodide-0.19.0/packages/modbampy/build/modbampy-0.4.0/htslib/../.git/modules/htslib
rm -f test/*.tmp test/*.tmp.* test/longrefs/*.tmp.* test/tabix/*.tmp.* test/tabix/FAIL* header-exports.txt shlib-exports-so.txt
rm -rf htscodecs/tests/test.out
rm -f *.o *.pico cram/*.o cram/*.pico test/*.o test/*.dSYM config_vars.h version.h
rm -f htscodecs/htscodecs/*.o htscodecs/htscodecs/*.pico htscodecs/htscodecs/version.h
rm -f hts-object-files
rm -f htscodecs/tests/*.o
rm -f libhts.so libhts.so.*
rm -f libhts.a bgzip htsfile tabix  test/hts_endian test/fieldarith test/hfile test/pileup test/pileup_mod test/plugins-dlhts test/sam test/test_bgzf test/test_expr test/test_kfunc test/test_kstring test/test_mod test/test_realn test/test-regidx test/test_str2int test/test_view test/test_index test/test-vcf-api test/test-vcf-sweep test/test-bcf-sr test/fuzz/hts_open_fuzzer.o test/test-bcf-translate test/test-parse-reg test/test_introspection test/thrash_threads1 test/thrash_threads2 test/thrash_threads3 test/thrash_threads4 test/thrash_threads5 test/thrash_threads6 test/thrash_threads7
rm -f htscodecs/tests/rans4x8 htscodecs/tests/rans4x16pr htscodecs/tests/arith_dynamic htscodecs/tests/tokenise_name3 htscodecs/tests/fqzcomp_qual htscodecs/tests/varint
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for ranlib... ranlib
checking for grep that handles long lines and -e... /usr/bin/grep
checking for C compiler warning flags... -Wall
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking shared library type for unknown-Linux... plain .so
checking whether the compiler accepts -fvisibility=hidden... yes
checking how to run the C preprocessor... gcc -E
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/param.h... yes
checking for getpagesize... yes
checking for working mmap... yes
checking for gmtime_r... yes
checking for fsync... yes
checking for drand48... yes
checking for srand48_deterministic... no
checking whether fdatasync is declared... yes
checking for fdatasync... yes
checking for library containing log... -lm
checking for zlib.h... yes
checking for inflate in -lz... yes
checking for library containing recv... none required
checking for bzlib.h... yes
checking for BZ2_bzBuffToBuffCompress in -lbz2... yes
checking whether htscodecs files are present... yes
checking for libdeflate.h... no
checking for libdeflate_deflate_compress in -ldeflate... no
configure: WARNING: GCS support not enabled: requires libcurl support
configure: WARNING: S3 support not enabled: requires libcurl support
checking for library containing regcomp... none required
checking whether PTHREAD_MUTEX_RECURSIVE is declared... yes
configure: creating ./config.status
config.status: creating config.mk
config.status: creating htslib.pc.tmp
config.status: creating config.h
config.status: config.h is unchanged
config.status: linking htscodecs_bundled.mk to htscodecs.mk
fatal: not a git repository: /home/roham/software/pyodide-0.19.0/packages/modbampy/build/modbampy-0.4.0/htslib/../.git/modules/htslib
fatal: not a git repository: /home/roham/software/pyodide-0.19.0/packages/modbampy/build/modbampy-0.4.0/htslib/../.git/modules/htslib
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o kfunc.o kfunc.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o kstring.o kstring.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o bgzf.o bgzf.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o errmod.o errmod.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o faidx.o faidx.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o header.o header.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o hfile.o hfile.c
echo '#define HTS_VERSION_TEXT "1.14"' > version.h
echo '#define HTS_CC "emcc"' > config_vars.h
echo '#define HTS_CPPFLAGS ""' >> config_vars.h
echo '#define HTS_CFLAGS "-Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden"' >> config_vars.h
echo '#define HTS_LDFLAGS "-L/home/roham/anaconda3/envs/py395/lib -fvisibility=hidden "' >> config_vars.h
echo '#define HTS_LIBS "-lbz2 -lz -lm  "' >> config_vars.h
fatal: not a git repository: /home/roham/software/pyodide-0.19.0/packages/modbampy/build/modbampy-0.4.0/htslib/htscodecs/../../.git/modules/htslib/modules/htscodecs
Updating htscodecs/htscodecs/version.h : #define HTSCODECS_VERSION_TEXT ""
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o hts.o hts.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o hts_expr.o hts_expr.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o hts_os.o hts_os.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o md5.o md5.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o multipart.o multipart.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o probaln.o probaln.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o realn.o realn.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o regidx.o regidx.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o region.o region.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o sam.o sam.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o vcf_sweep.o vcf_sweep.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o tbx.o tbx.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o textutils.o textutils.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o thread_pool.o thread_pool.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o vcf.o vcf.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o vcfutils.o vcfutils.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_decode.o cram/cram_decode.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_encode.o cram/cram_encode.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_external.o cram/cram_external.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_index.o cram/cram_index.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_io.o cram/cram_io.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/cram_stats.o cram/cram_stats.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/mFILE.o cram/mFILE.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o cram/string_alloc.o cram/string_alloc.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/arith_dynamic.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/fqzcomp_qual.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/htscodecs.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/pack.o htscodecs/htscodecs/pack.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static4x16pr.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rANS_static.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/rle.o htscodecs/htscodecs/rle.c
emcc -Wall -I/home/roham/anaconda3/envs/py395/include -fvisibility=hidden -I.  -c -o htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/tokenise_name3.c
emar -rc libhts.a kfunc.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o header.o hfile.o hts.o hts_expr.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o region.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_stats.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/string_alloc.o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/pack.o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rle.o htscodecs/htscodecs/tokenise_name3.o  
ranlib libhts.a
/home/roham/anaconda3/envs/py395/lib/python3.9/site-packages/cffi/cparser.py:154: UserWarning: String literal found in cdef() or type source. String literals are ignored here, but you should remove them anyway because some character sequences confuse pre-parsing.
  warnings.warn("String literal found in cdef() or type source. "
running install
/home/roham/anaconda3/envs/py395/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/home/roham/anaconda3/envs/py395/lib/python3.9/site-packages/setuptools/command/easy_install.py:156: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running bdist_egg
running egg_info
warning: no files found matching 'LICENSE.md'
no previously-included directories found matching 'htslib/test'
no previously-included directories found matching 'htslib/htscodecs/tests'
no previously-included directories found matching 'build'
no previously-included directories found matching 'docs'
writing manifest file 'modbampy.egg-info/SOURCES.txt'
running install_lib
running build_py
running build_ext
cc -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /home/roham/anaconda3/envs/py395/include -fPIC -O2 -isystem /home/roham/anaconda3/envs/py395/include -fPIC -Isrc -Ihtslib -I/home/roham/anaconda3/envs/py395/include/python3.9 -c build/temp.linux-x86_64-3.9/libmodbampy.c -o build/temp.linux-x86_64-3.9/build/temp.linux-x86_64-3.9/libmodbampy.o -std=c99 -msse3 -O3
cc -shared -Wl,-rpath,/home/roham/anaconda3/envs/py395/lib -Wl,-rpath-link,/home/roham/anaconda3/envs/py395/lib -L/home/roham/anaconda3/envs/py395/lib -Wl,-rpath,/home/roham/anaconda3/envs/py395/lib -Wl,-rpath-link,/home/roham/anaconda3/envs/py395/lib -L/home/roham/anaconda3/envs/py395/lib build/temp.linux-x86_64-3.9/build/temp.linux-x86_64-3.9/libmodbampy.o pymod.a htslib/libhts.a -lm -lz -llzma -lbz2 -lpthread -lcurl -lcrypto -o build/lib.linux-x86_64-3.9/libmodbampy.abi3.so
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/modbampy
byte-compiling build/bdist.linux-x86_64/egg/modbampy/__init__.py to __init__.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/libmodbampy.py to libmodbampy.cpython-39.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/entry_points.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/not-zip-safe -> build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying modbampy.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
removing '/home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python/modbampy-0.4.0-py3.9-linux-x86_64.egg' (and everything under it)
creating /home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python/modbampy-0.4.0-py3.9-linux-x86_64.egg
Extracting modbampy-0.4.0-py3.9-linux-x86_64.egg to /home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python
byte-compiling /home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python/modbampy-0.4.0-py3.9-linux-x86_64.egg/libmodbampy.py to libmodbampy.cpython-39.pyc
byte-compiling /home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python/modbampy-0.4.0-py3.9-linux-x86_64.egg/modbampy/__init__.py to __init__.cpython-39.pyc
modbampy 0.4.0 is already the active version in easy-install.pth
Installing modbampy script to /home/roham/software/pyodide-0.19.0/packages/.artifacts/bin

Installed /home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python/modbampy-0.4.0-py3.9-linux-x86_64.egg
Processing dependencies for modbampy==0.4.0
Searching for numpy==1.22.1
Best match: numpy 1.22.1
Processing numpy-1.22.1-py3.9-linux-x86_64.egg
numpy 1.22.1 is already the active version in easy-install.pth
Installing f2py script to /home/roham/software/pyodide-0.19.0/packages/.artifacts/bin
Installing f2py3 script to /home/roham/software/pyodide-0.19.0/packages/.artifacts/bin
Installing f2py3.9 script to /home/roham/software/pyodide-0.19.0/packages/.artifacts/bin

Using /home/roham/software/pyodide-0.19.0/packages/.artifacts/lib/python/numpy-1.22.1-py3.9-linux-x86_64.egg
Searching for cffi==1.15.0
Best match: cffi 1.15.0
Adding cffi 1.15.0 to easy-install.pth file

Using /home/roham/anaconda3/envs/py395/lib/python3.9/site-packages
Searching for pycparser==2.21
Best match: pycparser 2.21
Adding pycparser 2.21 to easy-install.pth file

Using /home/roham/anaconda3/envs/py395/lib/python3.9/site-packages
Finished processing dependencies for modbampy==0.4.0
[line 1 of 11]
emcc -O2 -g0 -fPIC -c -Wall -fstack-protector-strong -D_FORTIFY_SOURCE=2 -fpic -msse3 -std=c99 -Isrc -Ihtslib src/common.c -o common.o
emcc: error: Passing any of -msse, -msse2, -msse3, -mssse3, -msse4.1, -msse4.2, -mavx, -mfpu=neon flags also requires passing -msimd128!
```

Is the -msimd128 necessary somewhere that I can add?

Thanks!

**Comment by hoodmane** [comment]
You should filter out the `-msse3` flag in pywasmcross here:
https://github.com/pyodide/pyodide/blob/de61d5ab1f202ef6bcc522cf924e771dca726537/pyodide-build/pyodide_build/pywasmcross.py#L376
You could try adding `-msimd128` to `cflags` in `meta.yaml`, but I think we are not using simd for anything at this point.

**Comment by stevenweaver** [comment]
@rrazaghi -- Did you ever get pysam to work? Also having troubles.

**Comment by ryanking13** [comment]
Resolved by #4268. Thanks @stevenweaver!
